contracts/components/node_runners/NodeRunnerRegistryCore.sol

```diff
@@ -0,0 +1,379 @@
+    }
+
+    /**
+     * @notice gets the Scanner Node data struct at index registered to nodeRunnerId
+     * Useful for external iteration.
+     * @param nodeRunnerId ERC721 token id of the Node Runner.
+     * @param index of the registered Scanner Node. Must be lower than totalScannersRegistered(nodeRunnerId)
+     */
+    function registeredScannerAddressAtIndex(uint256 nodeRunnerId, uint256 index) external view returns (address) {
+        return _scannerNodeOwnership[nodeRunnerId].at(index);
+    }
+
+    // ************* Converters *************
+
+    /// Converts scanner address to uint256 for FortaStaking Token Id.
+    function scannerAddressToId(address scanner) public pure returns (uint256) {
+        return uint256(uint160(scanner));
+    }
+
+    /// Converts scanner address to uint256 for FortaStaking Token Id.
+    function scannerIdToAddress(uint256 scannerId) public pure returns (address) {
+        return address(uint160(scannerId));
+    }
+
+    // ************* Scanner Disabling *************
+
+    /// Gets if the disabled flag has been set for a Scanner Node Address
+    function isDisabled(address scanner) public view returns (bool) {
+        return _scannerNodes[scanner].disabled;
+    }
+
+    /**
+     * @notice Checks if the Scanner Node is considered enabled by the Forta Network, and is thus eligible for bot (Agent) assignment.
+     * @param scanner address
+     * @return true if:
+     * - Scanner Node is registered AND
+     * - Scanner Node's disabled flag is not set (is false) AND
+     * - (Scanner Node has more than minimum stake allocated to it OR staking is not enabled for the Scanner Node's chain)
+     */
+    function isEnabled(address scanner) public view returns (bool) {
+        return !_scannerNodes[scanner].disabled && _isStakedOverMin(scannerAddressToId(scanner));
+    }
+
+    /**
+     * @notice Checks if sender or meta-tx sender is allowed to set disabled flag for a Scanner Node
+     * @param scanner address
+     * @return true if _msgSender() is the NodeRunner owning the Scanner or the Scanner Node itself
+     */ 
+    function _canSetEnableState(address scanner) internal view virtual returns (bool) {
+        return _msgSender() == scanner || ownerOf(_scannerNodes[scanner].nodeRunnerId) == _msgSender();
+    }
+
+    /**
+     * @notice Sets Scanner Node disabled flag to false. It's not possible to re-enable a Scanner Node
+     * if the active stake allocated to it is below minimum for the scanned chainId. 
+     * If that happens, allocate more stake to it, then try enableScanner again.
+     * @param scanner address
+     */
+    function enableScanner(address scanner) onlyRegisteredScanner(scanner) public {
+        if (!_canSetEnableState(scanner)) revert CannotSetScannerActivation();
+        uint256 scannerId = scannerAddressToId(scanner);
+        if (!_isStakedOverMin(scannerId)) revert StakedUnderMinimum(scannerId);
```

Comment:
Is this strictly necessary? I would remove it for simplicity, since a token can be manually enabled and become ~disabled~ non-operational later anyway.

Reply:
Sorry, what in particular is strictly necessary?

Reply:
I meant making this function revert if the scanner isn't staked over min.

Reply:
We wanted to highlight the need for staking, since it may be confusing why enabling a scanner does not mean `isEnabled` is true. With `isOperational` it may be not needed

Reply:
82b16c3